src/drivers/cmakefileapi/api_helpers.ts

```diff
@@ -0,0 +1,244 @@
+import * as api from '@cmt/api';
+import * as cache from '@cmt/cache';
+import * as index_api from '@cmt/drivers/cmakefileapi/api';
+import {
+  CodeModelConfiguration,
+  CodeModelContent,
+  CodeModelFileGroup,
+  CodeModelProject,
+  CodeModelTarget
+} from '@cmt/drivers/codemodel-driver-interface';
+import * as logging from '@cmt/logging';
+import {fs} from '@cmt/pr';
+import rollbar from '@cmt/rollbar';
+import * as path from 'path';
+
+const log = logging.createLogger('cmakefileapi-helper');
+
+export async function createQueryFileForApi(api_path: string): Promise<string> {
+  const query_path = path.join(api_path, 'query', 'client-vscode');
+  const query_file_path = path.join(query_path, 'query.json');
+  await fs.mkdir_p(query_path);
+
+  const requests
+      = {requests: [{kind: 'cache', version: 2}, {kind: 'codemodel', version: 2}, {kind: 'cmakeFiles', version: 1}]};
+
+  await fs.writeFile(query_file_path, JSON.stringify(requests));
+  return query_file_path;
+}
+
+export async function loadIndexFile(reply_path: string): Promise<index_api.Index.IndexFile|null> {
+  log.debug(`Read reply folder: ${reply_path}`);
+  if (!await fs.exists(reply_path)) {
+    return null;
+  }
+
+  const files = await fs.readdir(reply_path);
+  log.debug(`Found index files: ${JSON.stringify(files)}`);
+
+  const index_file = files.find(filename => filename.startsWith('index-'));
+  if (!index_file) {
+    throw Error('Unexpected count of index files');
+  }
+  const index_file_path = path.join(reply_path, index_file);
+  const file_content = await fs.readFile(index_file_path);
+
+  return JSON.parse(file_content.toString()) as index_api.Index.IndexFile;
+}
+
+export async function loadCacheContent(filename: string): Promise<Map<string, api.CacheEntry>> {
+  const file_content = await fs.readFile(filename);
+  const cache_from_cmake = JSON.parse(file_content.toString()) as index_api.Cache.CacheContent;
+
+  return convertFileApiCacheToExtensionCache(cache_from_cmake);
+}
+
+function findPropertyValue(cacheElement: index_api.Cache.CMakeCacheEntry, name: string): string {
+  const property_element = cacheElement.properties.find(prop => prop.name == name);
+  return property_element ? property_element.value : '';
+}
+
+function convertFileApiCacheToExtensionCache(cache_from_cmake: index_api.Cache.CacheContent):
+    Map<string, api.CacheEntry> {
+  return cache_from_cmake.entries.reduce((acc, el) => {
+    const entry_type_translation_map: {[key: string]: api.CacheEntryType|undefined;} = {
+      BOOL: api.CacheEntryType.Bool,
+      STRING: api.CacheEntryType.String,
+      PATH: api.CacheEntryType.Path,
+      FILEPATH: api.CacheEntryType.FilePath,
+      INTERNAL: api.CacheEntryType.Internal,
+      UNINITIALIZED: api.CacheEntryType.Uninitialized,
+      STATIC: api.CacheEntryType.Static,
+    };
+    const type = entry_type_translation_map[el.type];
+    if (type === undefined) {
+      rollbar.error(`Unknown cache entry type ${el.type}`);
```

Comment:
I see this use of rollbar here, but not around any of the file operations.  Is it intentional that we let those exceptions bubble up to the top?